labs/b9.md

```diff
@@ -0,0 +1,178 @@
+
+You can see all of the images you’ve built on your machine with the `docker images` command.    
+
+##### Questions    
+
+1. Run the image you just built. As before, don’t forget the `-it` flags, and you can exit using CTRL+D. What do you observe?    
+2. Write and build a `Dockerfile` based on `ubuntu:xenial` that installs the packages `fortune` and `fortunes-min` and runs the fortune executable (located in `/usr/games/fortune` after you install it). Note that you won’t need to use the `-it` flags when you run the container as fortune doesn’t need `STDIN`. Submit your `Dockerfile` with this lab.  Hint: if you’re having trouble writing your Dockerfile, try booting an interactive container and installing both packages. How can you translate what you did interactively to a Dockerfile?   
+3. Paste the output running `docker images` command after questions 1 and 2.      
+
+#### Dockerizing a Web Application    
+
+For our last trick, we’re going to use Docker to run multiple Apache web servers inside containers. 
+
+For simplicity, you will not have to write this `Dockerfile`. Go ahead and pull the `httpd` image from Docker Hub. Now, it’s your job to figure out how to run three instances of the Apache containers on your machine.           
+
+Docker creates a separate network for containers, so you will need to forward your host port to your container’s port (this is called [port forwarding](https://en.wikipedia.org/wiki/Port_forwarding), or port mapping). The container is listening on port 80 by default. It is your job to run each instance on ports 5000, 5001, and 5002.  I recommend running the containers in detached mode with the `-d` flag.
+
+Hints (if you want them): 
+- Don’t expect to see any output until you visit the web application from your host machine.
+- The `-p` flag takes in a colon separated pair of HOST\_PORT:CONTAINER\_PORT (it can actually accept a ton of more options, but don’t worry about that for now). 
+- You should be able to view the web application from your host machine by visiting <url_of_host_machine>:5000, assuming you don’t have anything else running on that port.      
+ 
+ ##### Questions
+ 1. While your three containerized Apache web servers are running in detached mode, paste the output of `docker ps`.
+ 2. Observe that in the output of `docker ps` each container has an associated container ID. Explain why containers have IDs rather than being named after the application, for example httpd. 
+ 3. Now go ahead and stop your containers (see slide 30). Paste the command you used to stop one of the containers. 
+
+Congratulations! You’ve successfully Dockerized and ran a web application with vastly different requirements and configuration than your host machine, without affecting your setup on your machine :) There’s a lot more about Docker and containers to learn about, but I hope this was enough to wrap your head around the basic concepts and get some experience working with it.        
```

Comment:
This isn't relevant anymore since we're just running apache. Probably want to reword this